runtime/src/in_mem_accounts_index.rs

```diff
@@ -281,6 +343,22 @@ impl<T: IndexValue> InMemAccountsIndex<T> {
     fn remove_if_slot_list_empty_entry(&self, entry: Entry<K, AccountMapEntry<T>>) -> bool {
         match entry {
             Entry::Occupied(occupied) => {
+                if occupied.get().disk_unknown() {
+                    // merge memory with disk
+                    let entry = occupied.get();
+                    let key = occupied.key();
+                    let disk_entry = self.load_account_entry_from_disk(key);
+                    match disk_entry {
+                        Some(disk_entry) => {
+                            entry.clear_disk_unknown();
+                            Self::merge_slot_lists(entry, disk_entry)
```

Comment:
would be insightful to have metrics on how many times we upserted and set disk_unknown, removed and it was set, called 'get' and it was set

Reply:
Sure. I will add it in a follow up pull request.

Reply:
Added.